drivers/i2s/i2s_mcux_flexcomm.c

```diff
@@ -19,7 +19,7 @@
 
```

Comment:
Ideally you should add only one patch per issue addressed. This way the code would be much easier to understand and review.

There are exceptions if the changes are tightly related. Not sure if this the case.

Reply:
The primary bug is the change/regression in the DMA driver. But the fix to the DMA driver includes an assert, which is there to prevent future mistakes from happening silently. That assert would immediately break the I2S driver, which is why the two commits are combined in this PR.

I could take out the I2S one, but it would have to be merged first. I'm still relatively new to Zephyr (and github) development so I don't know if pull requests can be dependent on one another. Let me know if there's still a preference in having the I2S change be a separate Pull Request.